src/Integration/VsShellUtils.cs

```diff
@@ -118,5 +147,42 @@ public static bool SaveSolution(IServiceProvider serviceProvider, bool silent)
             // True if user clicked Yes, false otherwise (No/Cancel/Esc/Close dialog)
             return hr != VSConstants.E_ABORT && ErrorHandler.ThrowOnFailure(hr) == VSConstants.S_OK;
         }
+
+
+        #region Output pane helpers
+
+        public static IVsOutputWindowPane GetOrCreateSonarLintOutputPane(IServiceProvider serviceProvider)
+        {
+            IVsOutputWindow outputWindow = serviceProvider.GetService<SVsOutputWindow, IVsOutputWindow>();
+            if (outputWindow == null)
+            {
+                Debug.Fail("Could not get IVsOutputWindow");
+                return null;
+            }
+
+            const bool makeVisible = true;
+            const bool clearWithSolution = true;
+
+            IVsOutputWindowPane pane;
+            int hrCreatePane = outputWindow.CreatePane(
+                ref SonarLintOutputPaneGuid,
+                Strings.SonarLintOutputPaneTitle,
+                Convert.ToInt32(makeVisible),
+                Convert.ToInt32(clearWithSolution));
+            Debug.Assert(ErrorHandler.Succeeded(hrCreatePane), "Failed in outputWindow.CreatePane: " + hrCreatePane.ToString());
+                
+            int hrGetPane = outputWindow.GetPane(ref SonarLintOutputPaneGuid, out pane);
+            Debug.Assert(ErrorHandler.Succeeded(hrGetPane), "Failed in outputWindow.GetPane: " + hrGetPane.ToString());
+
+            return pane;
+        }
+
+        private static void WriteLineToPane(IVsOutputWindowPane pane, string messageFormat, params object[] args)
+        {
+            int hr = pane.OutputStringThreadSafe(Environment.NewLine + string.Format(CultureInfo.CurrentCulture, messageFormat, args: args));
```

Comment:
You're pre-pending the newline here but post-pending it from the notifier class that is also writing to the same pane. Isn't that going to cause some formatting issues?